src/nginx_config.py

```diff
@@ -261,6 +265,7 @@ def _build_server_config(
                     "args": ["X-Scope-OrgID", "$ensured_x_scope_orgid"],
                 },
                 {"directive": "server_name", "args": [self.server_name]},
+                *self._resolver(custom_resolver=_get_dns_ip_address()),
```

Comment:
It's not added when TLS is on - you also need to add it in line 249. Probably the same problem exists in all other tandem PRs.